tests/RobotFramework/tests/cumulocity/availability/heartbeat.robot

```diff
@@ -0,0 +1,84 @@
+*** Settings ***
+Resource    ../../../resources/common.resource
+Library    Cumulocity
+Library    ThinEdgeIO
+
+Test Tags    theme:c8y
+Test Setup    Test Setup
+Test Teardown    Get Logs
+
+*** Test Cases ***
+
+### Main Device ###
+Heartbeat is sent
+    # How to test if a heartbeat is sent?
+    # c8y has time lag to change the availability state
+    # that may cause false-negative result if we check c8y_Availability.status
+    # also putting 'sleep 2m' a couple of times is not good :/
+
```

Comment:
What about replacing Sleep commands with loops that periodically check the desired status. This approach can reduce wait times and improve test reliability. for Example something like:
`Wait Until Keyword Succeeds    2m    10s    Device Should Have Fragment Values    c8y_Availability.status=AVAILABLE`